pkg/clients/etcd.go

```diff
@@ -0,0 +1,40 @@
+package clients
+
+import (
+	"fmt"
+	"time"
+
+	logging "github.com/op/go-logging"
+
+	"github.com/coreos/etcd/client"
+)
+
+type EtcdConfig struct {
```

Comment:
> Speaking of clients.() methods, do we want any of these to be singletons?

This sounds ok, we need to make sure we test multiple users of the SAME client at the SAME time. Will that work ok? IS it really a 'single' connection? The old way we created a new connection so if there were multiple provisions going on each worker would have their own connection.